Tests/TSCBasicTests/FileSystemTests.swift

```diff
@@ -884,22 +783,31 @@ class FileSystemTests: XCTestCase {
         let fileB = path.appending(component: "fileB")
         let lockFile = path.appending(component: "lockfile")
 
+        try _testFileSystemFileLock(fileSystem: fs, fileA: fileA, fileB: fileB, lockFile: lockFile)
+    }
+
+    private func _testFileSystemFileLock(fileSystem fs: FileSystem, fileA: AbsolutePath, fileB: AbsolutePath, lockFile: AbsolutePath) throws {
+
+        // write initial value, since reader may start before writers and files would not exist
+        try fs.writeFileContents(fileA, bytes: [0])
```

Comment:
This is writing a null byte, so the code below that tries to read an integer as a string from the file will still fail.  I guess it's "okay" in the sense that it will then hit the `?? 0`, but it seems kind of indirect and hard to follow, especially since it's easy to misread writing 0 as being equivalent to writing "0".  Is there a reason not to write "0" to the file instead of "\0" and then simplify the logic below to force the optional Int?

Reply:
good point, I can revise